homework/hw36/hw36.js

```diff
@@ -0,0 +1,261 @@
+    event.preventDefault();
+    if (validateForm()) {
+        orderForm.style.display = "none";
+        orderConfirmation.style.display = "block";
+        displayOrderConfirmation();
+    }
+}
+
+function validateForm() {
+    const name = document.getElementById("name-input").value.trim();
+    const city = document.getElementById("city-select").value.trim();
+    const postOffice = document.getElementById("post-office-input").value.trim();
+    const payment = document.querySelector('input[name="payment"]:checked');
+    const quantity = document.getElementById("quantity-input").value.trim();
+
+    if (!name || !city || !postOffice || !payment || !quantity) {
+        const errorDiv = document.createElement("div");
+        errorDiv.style.color = "red";
+        errorDiv.textContent = "Please fill in all required fields.";
+        document.querySelector("form").appendChild(errorDiv);
+        return false;
+    }
+
+    return true;
+}
+
+function displayOrderConfirmation() {
+    const p = document.getElementById("product-name");
+    const productName = p.innerHTML.slice(8, -1);
+    const product = productsByName[productName];
+
+    const name = document.getElementById("name-input").value.trim();
+    const city = document.getElementById("city-select").value.trim();
+    const postOffice = document.getElementById("post-office-input").value.trim();
+    const payment = document.querySelector('input[name="payment"]:checked').value;
+    const quantity = document.getElementById("quantity-input").value.trim();
+    const comment = document.getElementById("comment-input").value.trim();
+
+    const productInfoText = `
+        Product: ${productName},
+        Quantity: ${quantity}`;
+    const deliveryInfoText = `
+        Delivery to: ${name}, 
+        ${city}, Post Office Nr: ${postOffice}, 
+        Payment method: ${payment},
+        Comment: ${comment}
+        `;
+
+    const productInfo = document.getElementById("product-info");
+    const deliveryInfo = document.getElementById("delivery-info");
+    productInfo.textContent = productInfoText;
+    deliveryInfo.textContent = deliveryInfoText;
+
+    product.date = new Date();
+    product.quantity = +quantity;
+    product.price *= product.quantity
+    console.log(product);
+
+    localStorage.setItem(`${product['id']}`, JSON.stringify(product));
+
+}
+
+const purchaseListBtn = document.querySelector('#myPurchase');
+const purchaseList = document.querySelector('.purchaseList');
+const categoriesContainer = document.querySelector('.container');
+const orderList = document.querySelector('#order-list');
+const totalPriceDiv = document.querySelector('#totalPrice')
+
+purchaseListBtn.addEventListener('click', showPurchaseList);
+
+function showPurchaseList() {
+    categoriesContainer.classList.add('hidden');
+    purchaseListBtn.style.display = 'none';
+    purchaseList.style.display = 'block';
+
+    const orders = [];
+    let keys = Object.keys(localStorage);
```

Comment:
```
localStorage.setItem(`${product['id']}`, JSON.stringify(product));

// ...
let keys = Object.keys(localStorage);
```

You need to improve your solution, because somebody could write the wrong data and the application can fail

```
// Use constant key for your solution
const SOME_STORAGE_KEY = "STORAGE_KEY";

// ...

// Get products from LS, maybe saved before
const rawFromLS = localStorage.getItem(SOME_STORAGE_KEY);
const productsFromLS = rawFromLS == null ? {} : JSON.parse(rawFromLS);

productsFromLS[product.id] = product;

localStorage.setItem(SOME_STORAGE_KEY, JSON.stringify(productsFromLS));

// ...
const rawFromLS = localStorage.getItem(SOME_STORAGE_KEY);
const productsFromLS = rawFromLS == null ? {} : JSON.parse(rawFromLS);

const keys = Object.keys(productsFromLS);
```